src/main/java/hudson/plugins/promoted_builds/JobPropertyImpl.java

```diff
@@ -79,6 +79,13 @@ public JobPropertyImpl(AbstractProject<?,?> owner) throws Descriptor.FormExcepti
         init();
     }
 
+    /**
+     * Programmatic construction.
+     */
+    public JobPropertyImpl(Set<String> activeProcessNames) throws Descriptor.FormException, IOException {
+        this.activeProcessNames.addAll(activeProcessNames);
```

Comment:
This change is needed to reuse the class for the DSL plugin. Otherwise I would have to create a complete new class like JobDslPromotionProcess. But I tried to avoid that as often as possible to reduce maintenance overhead.